R/Data-methods.R

```diff
@@ -231,84 +231,75 @@ setMethod(
   }
 )
 
-## --------------------------------------------------
-## Update a Data object
-## --------------------------------------------------
-
-
-##' Update method for the "Data" class
-##'
-##' Add new data to the \code{\linkS4class{Data}} object
-##'
-##' @param object the old \code{\linkS4class{Data}} object
-##' @param x the dose level (one level only!)
-##' @param y the DLT vector (0/1 vector), for all patients in this cohort
-##' @param ID the patient IDs
-##' @param newCohort logical: if TRUE (default) the new data are assigned
-##' to a new cohort
-##' @param \dots not used
-##' @return the new \code{\linkS4class{Data}} object
-##'
-##' @example examples/Data-method-update-Data.R
-##' @export
-##' @keywords methods
-setMethod("update",
-          signature=
-          signature(object="Data"),
-          def=
-          function(object,
-                   x,
-                   y,
-                   ID=(if(length(object@ID)) max(object@ID) else 0L) + seq_along(y),
-                   newCohort=TRUE,
-                   ...){
-
-              ## some checks
-              stopifnot(is.scalar(x),
-                        all(y %in% c(0, 1)))
-
-              ## which grid level is the dose?
-              gridLevel <- matchTolerance(x, object@doseGrid)
-
-              ## add it to the data
-              if(is.na(gridLevel))
-              {
-                  stop("dose is not on grid")
-              } else {
-                  object@xLevel <- c(object@xLevel,
-                                     rep(gridLevel,
-                                         length(y)))
-              }
-
-              ## increment sample size
-              object@nObs <- object@nObs + length(y)
-
-              ## add dose
-              object@x <- c(object@x,
-                            rep(x,
-                                length(y)))
-
-              ## add DLT data
-              object@y <- c(object@y, as.integer(y))
-
-              ## add ID
-              object@ID <- c(object@ID, ID)
-
-              ## add cohort number
-              if(newCohort){
-                  object@cohort <- c(object@cohort,
-                                     rep(max(tail(object@cohort, 1L), 0L) + 1L,
-                                         length(y)))
-              }else{
-                  object@cohort <- c(object@cohort,
-                                     rep(max(tail(object@cohort, 1L), 0L),
-                                         length(y)))
-              }
+# Data-update ----
 
-              ## return the object
-              return(object)
-          })
+#' Update method for the [`Data`] class.
```

Comment:
```suggestion
#' Updating `Data` Objects
```
Title = noun + uppercase style